### DISCUSSION TITLE
Setup Debug service on PUB400

### DISCUSSION THREAD

**Discussion by JohnMP01** [codefori/vscode-ibmi#1938] [IBM i]
When on PUB400 I am getting the error below when I try to setup certificates (Generate Certificates)
  or try to Start up the service (Start Debug Service)

Failed to Create Certificate directory :
/QIBM/ProdData/IBMiDebugService/bin/certs

If I try to debug a program I get error:
EQAVS1007E PUB400.com on port 8005 could not be connected.
Message received: connect ECONNREFUSED 185.113.5.134:8005

Any help is appreciated.
Thanks
John

**Comment by sebjulliand** [comment]
Hi @JohnMP01 ,
it is not possible to debug from VSCode on PUB400, for multiple reasons:
- The debug server needs to be running on the LPAR; it can be started using the `STRDBGSVR` command, but someone with enough authority needs to do it.
- The certificate for the debug service must be generated once by an admin and be put in a publicly accessible location if secure debug is turned on.
- The debug service must be started once for everyone by someone with enough authority. Since this is a shared service, it must not be started by everyone.

In a nutshell, enabling the debug service must be done by a PUB400 admin and starting the debug server and debug service must be done (ideally) when the LPAR starts.

We had a discussion here about this if you want to know more: https://github.com/codefori/docs/pull/27

**Reply by JohnMP01** [reply]
Hi @sebjulliand , 

Thank you very much  for the update on the PUB400 situation. 
I am assuming the PUB400 Admins know about this and are hopefully considering what they can do to make the Debug Service available to Code for i users via VS code.
Maybe I should repost this on the forum located on PUB400 , to help get it resolved?

**Reply by sebjulliand** [reply]
You're welcome @JohnMP01 !
Looks like @chrjorgensen just sent a request to PUB400's admin about this: https://github.com/codefori/docs/pull/27#issuecomment-2002533586
